lib/vast/Conversion/Parser/ToParser.cpp

```diff
@@ -75,6 +78,141 @@ namespace vast::conv {
+            .filePath = std::nullopt,
+            .range = std::nullopt,
+        };
+
+        if (auto req_loc = get_location(loc)) {
+            req.filePath = req_loc->filePath;
+            req.range    = req_loc->range;
+        }
+
+        auto response = server.send_request(req);
+        if (auto result = std::get_if< vast::server::input_request::response_type >(&response))
+        {
+            if (result->value == "nonparser") {
+                return function_category::nonparser;
+            } else if (result->value == "sink") {
+                return function_category::sink;
+            } else if (result->value == "source") {
+                return function_category::source;
+            } else if (result->value == "parser") {
+                return function_category::parser;
+            }
+        }
+        return function_category::nonparser;
+    }
+
+    pr::data_type ask_user_for_return_type(vast::server::server_base &server, hl::FuncOp op) {
+        auto loc = op.getLoc();
+
+        vast::server::input_request req{
+            .type = { "maybedata", "nodata", "data" },
+            .text = "Please choose return type for function `" + op.getSymName().str() + '`',
+            .filePath = std::nullopt,
+            .range    = std::nullopt,
+        };
+
+        if (auto req_loc = get_location(loc)) {
+            req.filePath = req_loc->filePath;
+            req.range    = req_loc->range;
+        }
+
+        auto response = server.send_request(req);
+        if (auto result = std::get_if< vast::server::input_request::response_type >(&response))
+        {
+            return parse_type_name(result->value);
+        }
+        return pr::data_type::maybedata;
+    }
+
+    pr::data_type ask_user_for_argument_type(
+        vast::server::server_base &server, hl::FuncOp op, unsigned int idx
+    ) {
+        auto num_body_args = op.getFunctionBody().getNumArguments();
+
+        vast::server::input_request req{
+            .type = { "maybedata", "nodata", "data" },
+            .text = "Please choose a type for argument " + std::to_string(idx)
+                + " of function `" + op.getSymName().str() + '`',
+            .filePath = std::nullopt,
+            .range    = std::nullopt,
+        };
+
+        if (idx < num_body_args) {
+            auto arg = op.getArgument(idx);
+            auto loc = arg.getLoc();
+            if (auto req_loc = get_location(loc)) {
+                req.filePath = req_loc->filePath;
+                req.range    = req_loc->range;
+            }
+        }
+
+        auto response = server.send_request(req);
+        if (auto result = std::get_if< vast::server::input_request::response_type >(&response))
+        {
+            return parse_type_name(result->value);
+        }
+        return pr::data_type::maybedata;
+    }
+
+    function_model
+    ask_user_for_function_model(vast::server::server_base &server, hl::FuncOp op) {
```

Comment:
use `core::function_op_interface` everywhere instead of `hl::FuncOp`

Reply:
I can't find an equivalent of `getSymName()` for `FunctionOpInterface`

Reply:
`dyn_cast<SymbolOpInterface>(op.getOperation()).getName()`

Reply:
+ some checks you really have symbol